### ISSUE TITLE
Lazy-Load Option For Widgets (Like Dynamic Visibility)

### ISSUE THREAD

**Issue by itsinalis** [CrocoBlock/suggestions#4413]
I'm using jet plugins for over a year now. I had my objections at the beginning but the plugins are only getting better, making Crocoblock a framework of tools that you can rely on.

When it comes to caching a dynamic website you either have to sacrifice page speed in order to keep the pages displaying the data correctly, or to build pages with "less" dynamic features in order to be able to cache the pages without making the dynamic functionality static.

I'm facing this problem for months now, and the only thing I see as a solution is to lazy-load the dynamic content so it won't get cached together with the rest of the page, while the rest of the page (static content) will be loading instantly.

I've tested this solution and it works perfectly:

Since the only widget on Jet Engine that allows a lazy-load option is the Listing Grid Widget, I created Listing Items for each dynamic element on my pages, and using the Listing Grid Widget, I pulled the data on the frontend having the lazy-load option turned on.

The result was that my pages are loading in less than a second, and the dynamic content loads with a tiny delay but displaying all the information correctly, giving an excellent experience to the end-user.

In conclusion, it's not really practical nor efficient to build separate Listing Items for each dynamic content on the website, nor to buy an extra add-ons plugin just for the lazy-loading option.

It would be really helpful to have a Lazy-loading option in the advanced tab (or somewhere) just like the Dynamic Visibility toggle (and it would be a really nice addition to have the option to set a custom pre-loader lottie or gif).

I know for a fact that this will help a lot of users that want to build advanced websites using Elementor and Crocoblock.

I just purchased Crocoblock Unlimited and spent 30 minutes writing my request so I hope that you'll make me happy!

Kind regards

**Comment by terry2015edu** [comment]
+1
I am looking for this function for so long!
I really love the dynamic function JetEngine Provided. However, when running some complicated websites, lazyload widgets (instead of disabling page cache and load every time completely) is the best solution for dynamic sites

Please consider this Croco Team! Thank you!

**Comment by Lonsdale201** [comment]
+1

**Comment by terry2015edu** [comment]
+1
Finding the solution for so long.
Crocoblock Suite is really helpful for me to build a complex website with tons of dynamic data showing. However, I have to sacrifice the speed nor the UX (just like what itsinalis said). I tried many ways, like litespeed ESI, 3rd party addon for lazyload (totally messed with the function) and had no lucks.

BTW thanks to itsinalis for "listing grid inside listing grid". Never thought about this👏

**Comment by itsinalis** [comment]
> +1 Finding the solution for so long. Crocoblock Suite is really helpful for me to build a complex website with tons of dynamic data showing. However, I have to sacrifice the speed nor the UX (just like what itsinalis said). I tried many ways, like litespeed ESI, 3rd party addon for lazyload (totally messed with the function) and had no lucks.
> 
> BTW thanks to itsinalis for "listing grid inside listing grid". Never thought about this👏

**Comment by itsinalis** [comment]
@terry2015edu I've spent countless hours figuring out a solution to caching dynamic content! I'm happy to know I've helped you, and I really hope that @Crocoblock will make it easier and more efficient for us by implementing Lazy-Loading functionality to Jet Engine!

@MakhonkoDenis @MjHead
